astropy/utils/tests/test_data.py

```diff
@@ -279,19 +289,28 @@ def osraiser(dirnm, linkto):
     assert not os.path.isdir(lockdir), 'Cache dir lock was not released!'
 
 
-def test_read_unicode():
+@pytest.mark.parametrize(('filename'), ['unicode.txt', 'unicode.txt.gz', 'unicode.txt.bz2', 'unicode.txt.xz'])
+def test_read_unicode(filename):
     from ..data import get_pkg_data_contents
 
-    contents = get_pkg_data_contents('data/unicode.txt', encoding='utf-8')
-    assert isinstance(contents, six.text_type)
-    contents = contents.splitlines()[1]
-    assert contents == "האסטרונומי פייתון"
+    try:
+        contents = get_pkg_data_contents(os.path.join('data', filename), encoding='utf-8')
+        assert isinstance(contents, six.text_type)
+        contents = contents.splitlines()[1]
+        assert contents == "האסטרונומי פייתון"
 
-    contents = get_pkg_data_contents('data/unicode.txt', encoding='binary')
-    assert isinstance(contents, bytes)
-    x = contents.splitlines()[1]
-    assert x == b"\xff\xd7\x94\xd7\x90\xd7\xa1\xd7\x98\xd7\xa8\xd7\x95\xd7\xa0\xd7\x95\xd7\x9e\xd7\x99 \xd7\xa4\xd7\x99\xd7\x99\xd7\xaa\xd7\x95\xd7\x9f"[1:]
+        contents = get_pkg_data_contents(os.path.join('data', filename), encoding='binary')
+        assert isinstance(contents, bytes)
+        x = contents.splitlines()[1]
+        assert x == b"\xff\xd7\x94\xd7\x90\xd7\xa1\xd7\x98\xd7\xa8\xd7\x95\xd7\xa0\xd7\x95\xd7\x9e\xd7\x99 \xd7\xa4\xd7\x99\xd7\x99\xd7\xaa\xd7\x95\xd7\x9f"[1:]
 
+    except ValueError:
+        if not HAS_BZ2 and 'bz2' in filename:
+            pass
+        elif not HAS_XZ and 'xz' in filename:
```

Comment:
Perhaps instead of this it would be good to check for BZ2/LZMA support at the beginning of the test and either skip the test, or if you expect a ValueError, you could explicitly test that the correct error is raised when support is not available.  I think I'd prefer the former though.  Skipping makes clear that something isn't being tested, where as passing would test that the behavior is correct when BZ2/LZMA support is not available, but would hide the fact that support isn't really being tested.